### PR TITLE
Adding HTML widget to show the kpoint setting changes

### PR DESCRIPTION
**Pull request by AndresOrtegaGuerrero**
This PR adds a HTML displaying the mesh that is going to be use in the calculations based on the kpoints_distances used. This was one issue in #418

### CODE REVIEW

File: src/aiidalab_qe/app/configuration/advanced.py [hunk 1]
```diff
@@ -261,6 +265,53 @@ def reset(self):
             # reset the override checkbox
             self.override.value = False
 
+    def _display_mesh(self, _=None):
+        if self.input_structure is not None:
+            if self.kpoints_distance.value > 0:
+                mesh = self.create_kpoints_from_distance(
```
Comment (unkcpz) [inline src/aiidalab_qe/app/configuration/advanced.py:271]: I don't think you need a modified version of this function. It is a calcification in QE plugin and aiida now support passing regular data type and aiida will convert to aiida datatype for you. So here you can import the function from QE plugin, only need to take care of the output to the correct datatype.

File: src/aiidalab_qe/app/configuration/advanced.py [hunk 2]
```diff
@@ -261,6 +265,53 @@ def reset(self):
             # reset the override checkbox
             self.override.value = False
 
+    def _display_mesh(self, _=None):
+        if self.input_structure is not None:
```
Comment (unkcpz) [inline src/aiidalab_qe/app/configuration/advanced.py:269]:
```suggestion
        if self.input_structure is None:
        return
```
Then the left part of code in this function has no need to be indented.

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks! I have two minor requests. Since we have a mature test framework and get more experience on writing unit test for widgets can you add a test for this class and test the function added here.

File: src/aiidalab_qe/app/configuration/advanced.py (hunk 1 above)
Comment (AndresOrtegaGuerrero) [reply src/aiidalab_qe/app/configuration/advanced.py:271]: @unkcpz Thank you i will adapt the code and write the test

**Comment by AndresOrtegaGuerrero** [comment]
@unkcpz  Do you have any suggestion on why the test is failing? , locally the advanced test is passing

**Comment by unkcpz** [comment]
I think that is because you call calcfunction but you don’t have rabbitmq. Please try adding this section https://github.com/aiidalab/aiidalab-widgets-base/blob/2c29a9eac70e9f0773fd93165daccc14c304919b/.github/workflows/ci.yml#L113

File: tests/configuration/test_advanced.py [hunk 3]
```diff
@@ -114,3 +114,40 @@ def test_advanced_tot_charge_settings():
...
+
+    from aiidalab_qe.app.configuration.advanced import AdvancedSettings
+
+    StructureData = DataFactory("structure")
+
+    w = AdvancedSettings()
+
+    # Create a StructureData for AdvancedSettings (Silicon)
+
+    structure = StructureData(
```
Comment (unkcpz) [inline tests/configuration/test_advanced.py:131]:
```suggestion
    structure = orm.StructureData(
```
We usually have this in other packages.

**Review by unkcpz — CHANGES_REQUESTED** [review]
Just one minor request, all good then.

File: tests/configuration/test_advanced.py (hunk 3 above)
Comment (AndresOrtegaGuerrero) [reply tests/configuration/test_advanced.py]: no problems i will call the StructureData via from aiida import orm,

**Review by unkcpz — APPROVED** [review]
Thanks! all great. @AndresOrtegaGuerrero

**Pull request merged**